### PR TITLE
feat(connector): [Adyen] implement Online Banking Fpx for Adyen

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Online Banking Fpx is a Malaysian specific payment method.
```
"payment_method_type": "online_banking_fpx",
  "payment_method_data": {
    "bank_redirect": {
      "online_banking_fpx": {
        "issuer": "alliance_bank"
      }
    }
  }
```

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Tested it using hs-payments-test.netlify.app
Screen Shot 2023-07-01 at 9 30 22 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by jagan-jaya — DISMISSED** [review]
Please check VAS compatibility also

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -771,6 +812,29 @@ pub enum PaymentType {
     Vipps,
 }
 
+#[derive(Debug, Eq, PartialEq, Serialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum AdyenIssuerID {
```
[the remark this answers is not in the document]
Comment (AkshayaFoiger) [reply crates/router/src/connector/adyen/transformers.rs:827]: I am using this struct for online banking Thailand also, hence can we keep it generic.

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/tests/connectors/adyen_uk_ui.rs [hunk 2]
```diff
@@ -727,4 +744,10 @@ fn should_make_adyen_dana_payment_test() {
     tester!(should_make_adyen_dana_payment);
 }
 
+#[test]
+#[serial]
+fn should_make_adyen_online_banking_fpx_payment_test() {
+    tester!(should_make_adyen_online_banking_fpx_payment);
+}
+
 // https://hs-payments-test.netlify.app/paypal-redirect?amount=70.00&country=US&currency=USD&mandate_data[customer_acceptance][acceptance_type]=offline&mandate_data[customer_acceptance][accepted_at]=1963-05-03T04:07:52.723Z&mandate_data[customer_acceptance][online][ip_address]=127.0.0.1&mandate_data[customer_acceptance][online][user_agent]=amet%20irure%20esse&mandate_data[mandate_type][multi_use][amount]=700&mandate_data[mandate_type][multi_use][currency]=USD&apikey=dev_uFpxA0r6jjbVaxHSY3X0BZLL3erDUzvg3i51abwB1Bknu3fdiPxw475DQgnByn1z
```
Comment (NishantJoshi00) [inline crates/router/tests/connectors/adyen_uk_ui.rs:753]: Why is this comment present here? it also mentions a API_KEY

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
Other than that, looks good!

File: crates/router/tests/connectors/adyen_uk_ui.rs (hunk 2 above)
Comment (ArjunKarthik) [reply crates/router/tests/connectors/adyen_uk_ui.rs:753]: @srujanchikke will pick this up in a separate PR

**Review by NishantJoshi00 — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**